### PR TITLE
Main implementation

### PR DESCRIPTION
**Pull request by aldemeery**
### Added

- Basic layout.
- Main JS Model (Item).
- DOM Manipulation.

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team
### You have both done well so far :clap: However, you have a few issues to fix before the project can be approved.
- [x] Add webpack and configure it properly [see](https://github.com/microverseinc/linters-config/tree/master/javascript) 
- [x] Separate your code into modules 
mod
- [x] Separate application logic from the dom manipulation
logic
- [x] Make a to-do item editable for users
- [x] Make a to-do item assignable to a project 
### Make the changes and submit for another code review. Good luck!
- Happy coding!

By Raymond
Slack @iamraymond

File: scripts.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+    const toggle = document.createElement('a');
+    toggle.addEventListener('click', (e) => {
+      e.preventDefault();
+      item.toggle();
+      item.save();
+      render();
+    });
+    toggle.innerHTML = 'Toggle';
+
```
Comment (rOluochKe) [inline scripts.js:139]: ### separate the app dom manipulation from the logic

File: scripts.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    a.dataset.name = value;
+    a.innerHTML = `${value}<span class="tooltiptext">Right-click to delete</span>`;
+    a.addEventListener('contextmenu', deleteProject);
+    a.addEventListener('click', switchProject);
+
+    document.querySelector('.projects').appendChild(a);
+  }
+
+  e.target.reset();
+};
```
Comment (rOluochKe) [inline scripts.js:221]: ### Separate app dom manipulation from the logic

File: scripts.js (hunk 1 above)
Comment (aldemeery) [reply scripts.js]: Resolved

File: scripts.js (hunk 2 above)
Comment (aldemeery) [reply scripts.js]: Resolved

**Comment by aldemeery** [comment]
The ability to edit items has been added.
When a user clicks on the `Edit` link the form is filled with the item data, so the user can edit it however they want.

To-do items are actually assigned to projects by creating them under newly created projects.

Please check out the demo gif below, to see that in action.
demo

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi @aldemeery :hand: 
Nice job on this project and also a nice job implementing the requested changes.
But there still somethings that we need to change before moving on.

- [x] As it said in the requirements, your application should use localStorage to save todos between sessions. This functionality's not working right now.
Screenshot from 2020-08-14 09-06-03

- [x] Your .eslintrc file is different from the one [provided by Microverse](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json). We should not change this file. We need to delete the additional rule and the commented ones as well.

Kindly submit for code review again after making these changes.
Happy Coding :computer: 
**Sergio Torres**

**Comment by aldemeery** [comment]
Requested changes have been implemented

**Review by rOluochKe — APPROVED** [review]
Hello Team
### You got it right this time :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next project.
- Happy coding!

**Pull request merged**